wsmaster/che-core-api-installer/src/main/java/org/eclipse/che/api/installer/server/impl/LocalInstallerRegistry.java

```diff
@@ -37,39 +44,71 @@
  */
 @Singleton
 public class LocalInstallerRegistry implements InstallerRegistry {
+    private static final Logger LOG = LoggerFactory.getLogger(LocalInstallerRegistry.class);
+
     private final InstallerDao installerDao;
 
     /**
      * Primary registry initialization with shipped installers.
      */
     @Inject
-    public LocalInstallerRegistry(Set<Installer> installers, InstallerDao installerDao) throws InstallerException {
+    public LocalInstallerRegistry(Set<Installer> installers,
+                                  InstallerDao installerDao) throws InstallerException {
         this.installerDao = installerDao;
 
         for (Installer i : installers) {
-            InstallerImpl installer = new InstallerImpl(i);
+            doInit(installerDao, i);
+        }
+    }
 
-            try {
-                installerDao.create(installer);
-            } catch (InstallerAlreadyExistsException e) {
-                // ignore
-            }
+    private void doInit(InstallerDao installerDao, Installer i) throws InstallerException {
+        String installerKey = InstallerFqn.of(i).toKey();
+
+        try {
+            Version.validate(i.getVersion());
+        } catch (IllegalArgumentException e) {
```

Comment:
Mb then you can throw `InstallerException` by `Version.validate`? Also, I'd create `VersionValidator` class to avoid static calls.